pkg/node/bootstrap.go

```diff
@@ -187,10 +187,11 @@ func bootstrapNode(
 
 	storer := inmemstore.New()
 
-	retrieve := retrieval.New(swarmAddress, storer, p2ps, kad, logger, acc, pricer, tracer, o.RetrievalCaching, noopValidStamp)
+	retrieve, cleanup := retrieval.New(swarmAddress, storer, p2ps, kad, logger, acc, pricer, tracer, o.RetrievalCaching, noopValidStamp)
```

Comment:
Instead of adding a separate return argument for the closer,  we can just make the retrieval service implement the io.Closer and then add that to the list of closers. This doesnt change the API and I guess would reduce some of the test changes as well.

Same for the pushsync protocol. It seems like both should be implementing the Close to cleanup their internal state from now on. I think this is better and exposing the skiplist.Close method directly. If we need to add more stuff to cleanup later, implementing the closer is a better option.